__init__.py

```diff
@@ -48,11 +55,19 @@ def set_volume(self, percent=None):
         self.handle_volume_request(
             Message("mycroft.volume.get", {"percent": percent}))
 
+    def set_volume_by_gui(self, percent=None):
```

Comment:
can gui be made a flag in the existing set_volume method instead?

Reply:
nope, because you still want to have voice intent update the GUI separately to avoid the loop, if you put a flag for GUI it will still send the simulated get request which it anyways as it is doing now, keeping them separate is the only fix

- Keeping it in one function even with or without flag will result in
  -- User Request Volume Change by Voice:
     --- set_volume is called -> Simulated GUI get event -> GUI is updated (No Issues Here)
 
  -- User Changes Volume by GUI actions:
    --- GUI already sets value -> set_volume is called -> GUI is pushed into a set / get loop because of simulated event
 
- Keeping it separate:
  -- User Request Volume Change by Voice:
   --- set_volume is called -> Simulated GUI get event -> GUI is updated (No Issues Here)
 
  -- User Changes Volume by GUI actions:
    --- GUI already sets value -> set_volume_gui is called -> Only Mycroft Side is updated (No Issues Here)

Reply:
This is the same issue also affecting the slider on the drop-down menu, keeping this separate also fixes the funky behavior seen when changing volume from the drop-down

Reply:
I have instead added a right-handed property "set_by_gui" which is by default set to false, only if it's set by the mycroft.volume.set.gui event it will be true and not run the simulated function, this removes the duplicate set_volume_by_gui method

Reply:
thats exactly what i meant